File: Configuration/src/main/java/com/medecineWebApp/Configuration/models/Patient.java
package com.medecineWebApp.Configuration.models;


import com.medecineWebApp.Configuration.enums.EmployeeStatus;
import com.medecineWebApp.Configuration.enums.Gender;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.List;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

public class Patient  {
    private Long id;
    private String firstName;
    private String lastName;
    private String email;
    private String password;
    private String phone;
    private Gender gender;
    private LocalDate birthDate;
    private String address;
    private String city;
    private String state;
    private String zip;
    private String country;
    private EmployeeStatus status;
    private String ImageUrl;

    private List<Treatment> treatments;

    private List<Appointment> appointments;



}

File: Client/frontend/src/app/models/patient.ts
export interface IPatient {
     id?: number;
  firstName?: string;
  lastName?: string;
  age?: number;
  email?: string;
  phone?: string;
  address?: string;
  city?: string;
  state?: string;
  postalCode?: string;
  country?: string;
  dateOfBirth?: Date;
  gender?: string;
  bloodType?: string;
  avatar?: string;
  medicalHistory?: string;
  emergencyContact?: string;
  emergencyPhone?: string;
  status?: string; // Active/Inactive
  createdAt?: Date;
  updatedAt?: Date;
}

export class Patient implements IPatient {
constructor(
     public id?: number,
  public firstName?: string,
    public lastName?: string,
    public age?: number,
    public email?: string,
    public phone?: string,
    public address?: string,
    public city?: string,
    public state?: string,
    public postalCode?: string,
    public country?: string,
    public dateOfBirth?: Date,
    public gender?: string,
    public bloodType?: string,
    public avatar?: string, 
    public medicalHistory?: string,
    public emergencyContact?: string,
    public emergencyPhone?: string,
    public status?: string, // Active/Inactive
    public createdAt?: Date,
    public updatedAt?: Date
 ) {}   
}